sources/procfs.go

```diff
@@ -675,6 +675,31 @@ func parseFileElements(path string, directoryDepth int) (name string, nodeName s
 	return name, nodeName, nil
 }
 
+func convertToBytes(s string) string {
+	new_s := ""
+	multiplier := 1
+	switch finalChar := s[len(s)-1:]; finalChar {
+	case "K":
+		new_s = strings.TrimSuffix(s, "K")
+		multiplier = 1000
+	case "M":
+		new_s = strings.TrimSuffix(s, "M")
+		multiplier = 1000000
+	case "G":
+		new_s = strings.TrimSuffix(s, "G")
+		multiplier = 1000000000
+	default:
+		//this should only happen for integers
+		return s
+	}
+	byteVal, err := strconv.Atoi(new_s)
+	if err != nil {
+		return s
+	}
+	byteVal = byteVal * multiplier
+	return strconv.Itoa(byteVal)
```

Comment:
This was the most egregious  mistake I made! The code was actually broken. Accounted for, though I'm not super crazy about a lot of the casting choices I made. I could force everything to float64 and minimize the casting, but that's the only simplification I see. Penny for your thoughts?